app/models/student.rb

```diff
@@ -38,6 +39,11 @@ def gpa
     student_number.to_s[-2..].to_f / 25
   end
 
+  def current_registration
+    @current_registration =
+      semester_registrations.where(semester: semester).last || semester_registrations.create
```

Comment:
Bir semester için birden fazla `semester_registrations` kaydı olabilir mi? Eğer olursa neden `last` kullanıyoruz?
Eğer bir semester için bir kayıt olacaksa `where(...).last` yerine `find_by` kullanmalıyız.